From d8b43e6880dbffcba08102b92f18a86e2dccbd00 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@qt.io>
Date: Tue, 29 Aug 2017 15:17:15 +0200
Subject: [PATCH] Fix multilib builds with different archs of toolchain and
 system

Set host_cpu to ensure GN gets on the right path matching the host of
the toolchain we give it in host_toolchain.

Task-number: QTBUG-62673
Change-Id: I99d6dfdbe4cee30bc240049b656d1ec73becce8a
Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>
---
 src/core/config/linux.pri | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/core/config/linux.pri b/src/core/config/linux.pri
index 714c864d..a5ad4c32 100644
--- a/src/core/config/linux.pri
+++ b/src/core/config/linux.pri
@@ -106,10 +106,14 @@ host_build {
 } else {
     gn_args += custom_toolchain=\"$$QTWEBENGINE_OUT_ROOT/src/toolchain:target\"
     gn_args += host_toolchain=\"$$QTWEBENGINE_OUT_ROOT/src/toolchain:host\"
+    GN_TARGET_CPU = $$gnArch($$QT_ARCH)
     cross_compile {
         gn_args += v8_snapshot_toolchain=\"$$QTWEBENGINE_OUT_ROOT/src/toolchain:v8_snapshot\"
-        GN_TARGET_CPU = $$gnArch($$QT_ARCH)
+        # FIXME: we should set host_cpu in case host-toolchain doesn't match os arch,
+        # but currently we don't it available at this point
         gn_args += target_cpu=\"$$GN_TARGET_CPU\"
+    } else {
+        gn_args += host_cpu=\"$$GN_TARGET_CPU\"
     }
     !contains(QT_CONFIG, no-pkg-config) {
         # Strip '>2 /dev/null' from $$pkgConfigExecutable()
